storybookjs/storybook
# Controls: Remove duplicate control IDs for improved accessibility

Closes #26144 

## What I did

Use random UUID instead of `row.key` as part of the control element ID

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run the default sandbox using `yarn start`
2. Modify one of the `.mdx` file, making sure it contains at least 2 stories with its own `Canvas` and `Controls` each. For example, add the following lines after `<Story of={ButtonStories.Basic} />` in `code/addons/docs/template/stories/docs2/MetaOf.mdx`
```
<Canvas of={ButtonStories.Basic} />
<Controls of={ButtonStories.Basic} />

<Story of={ButtonStories.One} />
<Canvas of={ButtonStories.One} />
<Controls of={ButtonStories.One} />
```
3. Access the modified story
4. Use inspect element to check the ID assigned to each control components

<img width="1440" alt="Screenshot 2024-02-28 at 10 09 22 PM" src="https://github.com/storybookjs/storybook/assets/11392022/4857dcf5-70c8-490e-9631-dfccffe0dfc3">
<img width="1440" alt="Screenshot 2024-02-28 at 10 09 11 PM" src="https://github.com/storybookjs/storybook/assets/11392022/5b9a37f4-ad64-42ce-befa-074392f9ad9c">



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
